src/json-enc/json-enc.js

```diff
@@ -1,12 +1,32 @@
-htmx.defineExtension('json-enc', {
-  onEvent: function(name, evt) {
-    if (name === 'htmx:configRequest') {
-      evt.detail.headers['Content-Type'] = 'application/json'
+(function() {
+  let api
+  htmx.defineExtension('json-enc', {
+    init: function(apiRef) {
+      api = apiRef
+    },
+
+    onEvent: function(name, evt) {
+      if (name === 'htmx:configRequest') {
+        evt.detail.headers['Content-Type'] = 'application/json'
+      }
+    },
+
+    encodeParameters: function(xhr, parameters, elt) {
+      xhr.overrideMimeType('text/json')
+
+      const vals = api.getExpressionVars(elt)
+      const object = {}
+      parameters.forEach(function(value, key) {
```

Comment:
This loop could be smaller using a for... of loop imo. Depends on preferences using DRY principles.